### ISSUE TITLE
`keep` for `Record` (produces a subset)

### ISSUE THREAD

**Issue by kylegoetz** [gcanti/fp-ts#1647]
## 🚀 Feature request

It's quite common to have to write something that extracts a defined set of props from a record like:

```typescript
declare const o: Record<...>
pipe(
    o, 
    map(({foo, bar}) => ({foo, bar}))
)
```

### Desired Behavior

It would be nice to have a function in the `Record` module (called, say, `keep` or `subrecord`) that does this nicely.

### Suggested Solution

I would define the typing of this function as

```typescript
declare const keep: <R extends Record<string, unknown>, KS extends keyof R>(...ks: NonEmptyArray<KS>) => (r: R) => Pick<R, KS>
```

and code might be

```typescript
const _ = (...ks: NonEmptyArray<string>) => R.filterWithIndex((k, v) => ks.includes(k))
```

But I can't seem to get the typing to work with the implementation:

```typescript
const keep: <R extends Record<string, unknown>, KS extends keyof R>(...ks: NonEmptyArray<KS>) => (r: R) => Pick<R, KS> = (...ks) => 
    R.filterWithIndex((k, v) => ks.includes(k)) // TS2322: Type '<B extends unknown>(fb: Record<string, B>) => Record<string, B>' is not assignable to type '(r: R) => Pick<R, KS>'.   Type 'Record<string, unknown>' is not assignable to type 'Pick<R, KS>'.
```

### Who does this impact? Who is this for?

I suppose it impacts a lot of people. This use case arises for me regularly when receiving a big object from the network and need to squeeze it into an interface representing a proper subset of the incoming data (like converting `{ label, value, children, ... }[]` into `{ label, value }[]`

### Describe alternatives you've considered

`R.filterWithIndex((k, v) => ks.includes(k))` accomplishes the goal. Just thought maybe a utility function that is explicitly for doing this common task might be nice. I understand if maintainers don't want to add bloat for something this straightforward.

### Your environment

| Software   | Version(s) |
| ---------- | ---------- |
| fp-ts      |       2.10.5     |
| TypeScript |        3.9.10    |

**Comment by samhh** [comment]
Looks like [fp-ts-std/Record::pick](https://samhh.github.io/fp-ts-std/modules/Record.ts.html#pick), named after the `Pick` type. The type signature is very similar to your proposal.

There it's just [implemented](https://github.com/samhh/fp-ts-std/blob/3e2096ba16c38a6d5a9389755b066f9b5ab6ad7f/src/Record.ts#L69) with a cheap loop and an assertion.

**Comment by kylegoetz** [comment]
Yes, that's what I was talking about. I'll leave the issue open since this is technically a feature request "add to fp-ts" and not a support issue "how can I do this?" So admin/owner/contrib can close it as "won't be implemented" if y'all want.
